### PR TITLE
Scripts for local file migrations. Closes #642.

### LINKED ISSUE #642: Migration scripts for deletions and name changes in gist 11
See issue #613 for a delineation of these changes. This does not include the datetime model change, which will be handled by providing examples without a script, as detailed in that issue.

### PR DESCRIPTION
**Pull request by Jamie-SA**
Migration scripts from v10.0 to v11.0

Closes #642.

I think I covered what we can in the queries
I wrote a very simple test file in `./input/` to be able to do some testing.

### CODE REVIEW

File: migration/v11.0/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Migrating to gist v11.0
+
+This file gives a quick overview of using the migration utilities provided
+by the gist team to help migrate from the previous version of gist. These
```
Comment (rjyounes) [inline migration/v11.0/README.md:4]: "...previous version of gist, specifically 10.0.0."

File: migration/v11.0/input/test.ttl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix ex: <https://data.ex.com/data/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:_X_1 rdf:type gist:ContemporaneousEvent .
+gist:ContemporaneousEvent ex:somePredicate ex:_X_2 .
+
+ex:_X_3  gist:usesTimeZoneStandard ex:_TZ_X .
+
+ex:_X_4 rdf:type gist:HumanInstant ;
```
Comment (rjyounes) [inline migration/v11.0/input/test.ttl:15]: In gist 10, the object of `hasActualStart` is a `TimeInstant` object, not a datetime literal.

File: migration/v11.0/detect_renamed.rq [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            } UNION {
+                graph ?g { ?s ?p ?renamed }
+            } UNION {
+                graph ?g { ?renamed ?p ?o }
+            }
+        }
+    }
+    bind(<urn:renamed-entity-validation-report> as ?report)
+    bind(CONCAT("WARNING: incorrectly named property gist:",
+        REPLACE(STR(?renamed), '^.*[/#]', ''), " was changed to gist:isUnderJurisdiction. ",
```
Comment (rjyounes) [inline migration/v11.0/detect_renamed.rq:42]: renamed to `gist:isUnderJurisdictionOf`.

File: migration/v11.0/detect_renamed_default.rq [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+#            } UNION {
+#                graph ?g { ?s ?p ?renamed }
+#            } UNION {
+#                graph ?g { ?renamed ?p ?o }
+            }
+        }
+    }
+    bind(<urn:renamed-entity-validation-report> as ?report)
+    bind(CONCAT("WARNING: incorrectly named property gist:",
+        REPLACE(STR(?renamed), '^.*[/#]', ''), " was changed to gist:isUnderJurisdiction. ",
```
Comment (rjyounes) [inline migration/v11.0/detect_renamed_default.rq:42]: isUnderJurisdictionOf

**Review by rjyounes — COMMENTED** [review]
You can also do an action_rename on:

hasOrderedMember -> hasMember.

It might be worth adding in the detect_removed error message that we will be providing sample triples to illustrate the model changes and how they will need to change their data.

Otherwise looks good to me.

**Comment by rjyounes** [comment]
@Jamie-SA Since you wrote this we have all deleted `baseConversionFactor` and `standardConversionFactor` and use `conversionFactor` for both. Even though we have a convergence of two predicates, we can include the transformations in the migration scripts. Could you please add this?

File: migration/v11.0/input/test.ttl (hunk 2 above)
Comment (Jamie-SA) [reply migration/v11.0/input/test.ttl:15]: I wasn't trying to write correct triples, I just wanted a file that had the terms that needed to be tested.

File: migration/v11.0/detect_renamed_default.rq (hunk 4 above)
Comment (Jamie-SA) [reply migration/v11.0/detect_renamed_default.rq:42]: Fixed.

**Comment by Jamie-SA** [comment]
I've still got a couple more things to do on this PR, but have to switch gears right now and come back to it later.

File: migration/v11.0/README.md (hunk 1 above)
Comment (rjyounes) [reply migration/v11.0/README.md]: Either way, I don't have strong feelings.

File: migration/v11.0/input/test.ttl (hunk 2 above)
Comment (Jamie-SA) [reply migration/v11.0/input/test.ttl]: Removed datetime literal.

**Comment by Jamie-SA** [comment]
> It might be worth adding in the detect_removed error message that we will be providing sample triples to illustrate the model changes and how they will need to change their data.

I'm not sure what specifically you were referencing. Try to make suggestions for everything that changed?
That sounds like something that should be in the release notes.

**Comment by rjyounes** [comment]
Yeah, never mind that last comment.

**Review by rjyounes — CHANGES_REQUESTED** [review]
Did you add the renaming of the conversion factor predicates I mentioned? I don't see it. Both `standardConversionFactor` and `baseConversionFactor` are renamed to `conversionFactor`.

Once that's done, I think these are good to go unless you want to do anything else.

**Comment by Jamie-SA** [comment]
Sorry, missed the conversion factor renaming. I've pushed another change to include that.

**Comment by Jamie-SA** [comment]
I think I am done now. Please review.

**Review by rjyounes — COMMENTED** [review]
This looks good to me. The only thing I would add to the readme is that the migrations scripts are not cumulative - I.e. someone on v9 must upgrade to v10 first and then v11. (That's what I was trying to get at when I suggested mentioning v10, but this is a more generic way to say it.)

**Comment by Jamie-SA** [comment]
I made a couple more changes to address RY's last comment.

**Review by rjyounes — APPROVED** [review]

File: migration/v11.0/detect_renamed.rq [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                graph ?g { ?s ?p ?renamed }
+            } UNION {
+                graph ?g { ?renamed ?p ?o }
+            }
+        }
+    }
+    bind(<urn:renamed-entity-validation-report> as ?report)
+    bind(CONCAT("WARNING: incorrectly named property gist:",
+        REPLACE(STR(?renamed), '^.*[/#]', ''), " was changed to gist:isUnderJurisdictionOf. ",
+        "The previous name did not match the defintion. We won't autmatically update it ",
```
Comment (pwin) [inline migration/v11.0/detect_renamed.rq:43]: typo - "autmatically"

File: migration/v11.0/detect_renamed_default.rq [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+#                graph ?g { ?s ?p ?renamed }
+#            } UNION {
+#                graph ?g { ?renamed ?p ?o }
+            }
+        }
+    }
+    bind(<urn:renamed-entity-validation-report> as ?report)
+    bind(CONCAT("WARNING: incorrectly named property gist:",
+        REPLACE(STR(?renamed), '^.*[/#]', ''), " was changed to gist:isUnderJurisdictionOf. ",
+        "The previous name did not match the defintion. We won't autmatically update it ",
```
Comment (pwin) [inline migration/v11.0/detect_renamed_default.rq:43]: typo - "autmatically"

Comment (Jamie-SA) [reply migration/v11.0/detect_renamed_default.rq]: Fixed.

File: migration/v11.0/detect_renamed.rq (hunk 5 above)
Comment (Jamie-SA) [reply migration/v11.0/detect_renamed.rq]: Fixed

**Review by pwin — APPROVED** [review]
Looks good

**Pull request merged**